### PR TITLE
[iOS] Fixed the crash occurred on CarouselView2 when deleting last one remaining item with loop as false

### PR DESCRIPTION
**Pull request by Ahamed-Ali**
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

### Root Cause of the issue

- When removing the last item in `CarouselView2` with Loop = false, the item count is updated with an additional +2. This logic is specific to the looping scenario (to handle duplicate items), but there was no handling for the non-looping case. As a result, the update produced invalid indices and caused the exception. 

### Description of Change

- Added proper handling for the non-looping case (Loop = false) to prevent exceptions when removing the last item in CarouselView2. 

### Reference

- https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Handlers/Items/iOS/LoopObservableItemsSource.cs#L25

### Issues Fixed

Fixes #31535 

### Regression PR's

- #29318
- #26868

### Tested the behaviour in the following platforms

- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Screenshot

| Before Issue Fix | After Issue Fix |
|----------|----------|
| <video src="https://github.com/user-attachments/assets/53817665-393c-4b52-bfc4-2555d7339477"> | <video src="https://github.com/user-attachments/assets/dbe6e111-0217-47e4-a2f5-f897923cc566"> |

### CODE REVIEW

File: src/Controls/tests/TestCases.HostApp/Issues/Issue31535.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.ObjectModel;
+
+namespace Maui.Controls.Sample.Issues;
+
+[Issue(IssueTracker.Github, 31535, "[iOS] Crash occured on CarouselView2 when deleting last one remaining item with loop as false", PlatformAffected.iOS)]
```
[the remark this answers is not in the document]
Comment (Ahamed-Ali) [reply src/Controls/tests/TestCases.HostApp/Issues/Issue31535.cs]: modified

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+
+public class Issue31535 : _IssuesUITest
+{
+    public Issue31535(TestDevice device) : base(device) { }
+
+    public override string Issue => "[iOS] Crash occured on CarouselView2 when deleting last one remaining item with loop as false";
```
[the remark this answers is not in the document]
Comment (Ahamed-Ali) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs]: modified

File: src/Controls/src/Core/Handlers/Items2/iOS/LoopObservableItemsSource2.cs [hunk 3]
```diff
@@ -22,6 +22,10 @@ public LoopObservableItemsSource2(IEnumerable itemSource, UICollectionViewContro
 
 		protected override NSIndexPath[] CreateIndexesFrom(int startIndex, int count)
 		{
+			if (!Loop)
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Handlers/Items2/iOS/LoopObservableItemsSource2.cs:25]: Here, can validate also edge cases:

`if (count < 0)`

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+public class Issue31535 : _IssuesUITest
+{
+    public Issue31535(TestDevice device) : base(device) { }
+
+    public override string Issue => "[iOS] Crash occurred on CarouselView2 when deleting last one remaining item with loop as false";
+
+    [Test]
+    [Category(UITestCategories.CarouselView)]
+    public void CarouselView2ShouldNotCrashOnRemoveLastItem()
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs]: Could include another test, similar but removing more than one time the last item?

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs (hunk 4 above)
Comment (Ahamed-Ali) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs]: The current test already performs the remove action twice, so it seems you might be referring to the same scenario. Could you please confirm if this is sufficient, or would you prefer that we add more items and invoke the remove function multiple times? @jsuarezruiz

File: src/Controls/src/Core/Handlers/Items2/iOS/LoopObservableItemsSource2.cs (hunk 3 above)
Comment (Ahamed-Ali) [reply src/Controls/src/Core/Handlers/Items2/iOS/LoopObservableItemsSource2.cs:25]: Addressed

File: src/Controls/tests/TestCases.HostApp/Issues/Issue31535.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                var label = new Label();
+                label.SetBinding(Label.TextProperty, ".");
+
+                border.Content = label;
+                return border;
+            })
+        };
+
+        // Create Button
+        var button = new Button
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue31535.cs]: Can include other button to add items?

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs (hunk 4 above)
Comment (jsuarezruiz) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31535.cs]: Something like:
 
 1. Start with empty collection
 2. Add single item - should not crash
 3. Add multiple items - should work correctly
 4. Remove all items - should handle gracefully

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.HostApp/Issues/Issue31535.cs (hunk 5 above)
Comment (Ahamed-Ali) [reply src/Controls/tests/TestCases.HostApp/Issues/Issue31535.cs]: Added the test covering scenarios.
1. Started with Empty colleciton
2.Added Single Item 
3.Removed single item
4.Added Multiple Items
5.Removed All Items 

@jsuarezruiz

**Review by jsuarezruiz — APPROVED** [review]
